### ISSUE TITLE
Bug: ignores property acts differently when adding rules property

### ISSUE THREAD

**Issue by DavideViolante** [eslint/eslint#19008]
### Environment

Node version: v20.17.0
npm version: 10.8.2
Local ESLint version: 9.12.0
Global ESLint version: none
Operating System: Ubuntu 24.04.1

### What parser are you using?

Default (Espree)

### What did you do?

Repo here: https://github.com/DavideViolante/eslintv9-bug

#### Reproduce:
1. clone the repo I made and run `npm run lint`
2. you will notice errors reported in the dist folder, that should be ignored <--- cant understand why
3. remove the rules property from the eslint.config file
4. run `npm run lint` again and the errors about dist folder are gone <--- cant understand why

### What did you expect to happen?

to not report errors of dist folder if it's ignored

### What actually happened?

reports error of dist folder

### Link to Minimal Reproducible Example

https://github.com/DavideViolante/eslintv9-bug

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Not sure if related to typescript-eslint or here. I will open issue there too if needed.

**Comment by mdjermanovic** [comment]
> ignores property acts differently when adding rules property

This is intended behavior.

If you want ESLint to ignore certain files/directories completely, `ignores` should be the only key in the configuration object:

https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores

When `ignores` is used in a configuration object that has other keys, it only means that that configuration object doesn't apply to files matched by those `ignores`. But the file is still linted, and other configuration objects may apply. Also, in this case, ignore patterns match only files, so `"dist"` has no effect.

https://eslint.org/docs/latest/use/configure/configuration-files#excluding-files-with-ignores

In your repro, `tseslint.config()` returns a config array in which all objects have `ignores`, but also have other keys, so the files are not globally ignored, and `dist` has no effect. When you remove `rules`, one of the config objects will have only `ignores` key and thus act as global ignores.

Assuming you want ESLint to completely ignore `dist` and `*.config.js`, the config should be:

```js
// @ts-check
const eslint = require("@eslint/js");
const tseslint = require("typescript-eslint");

module.exports = tseslint.config(
  {
    extends: [
      eslint.configs.recommended,
      ...tseslint.configs.recommended,
    ],
    rules: {
      'indent': ['error', 2],
    }
  },
  {
    ignores: ['dist', '*.config.js']
  }
);
```
